File: Tang/Tang/Content/Module/Login/ViewCtrl/TangLoginGuideViewCtrl.m
//
//  TangLoginGuideViewCtrl.m
//  Tang
//
//  Created by wwwbbat on 2017/9/16.
//  Copyright © 2017年 tiny. All rights reserved.
//

#import "TangLoginGuideViewCtrl.h"
#import "TangLoginNaviCtrl.h"
#import "TangSession.h"

@interface TangLoginGuideViewCtrl ()
@property (strong, nonatomic) TangLoginNaviCtrl *loginNaviCtrl;
@end

@implementation TangLoginGuideViewCtrl

- (void)viewDidLoad
{
    [super viewDidLoad];
}

- (void)logined
{
    [SVProgressHUD show];
    RunAfter(.4f, ^{
        [SVProgressHUD showSuccessWithStatus:@"登录成功"];
        [self dismiss];
    });
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier isEqualToString:@"TangLoginNaviCtrlSegueID"]) {
        self.loginNaviCtrl = [segue destinationViewController];
        ygweakify(self);
        self.loginNaviCtrl.didLogined = ^{
            ygstrongify(self);
            [self logined];
        };
    }
}

@end

File: Tang/Tang/AppCode/Module/Login/Ctrls/TangLoginGuideViewCtrl.swift
//
//  TangLoginGuideViewCtrl.swift
//  Tang
//
//  Created by Tiny on 2018/2/8.
//  Copyright © 2018年 tiny. All rights reserved.
//

import UIKit
import SVProgressHUD

class TangLoginGuideViewCtrl: YGBasePopViewCtrl {

    private var loginNaviCtrl : TangLoginNaviCtrl!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    private func logined() {
        SVProgressHUD.show()
        RunAfter(0.4) {
            SVProgressHUD.showSuccess(withStatus: "登录成功")
            self.dismiss()
        }
    }

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let identifier = segue.identifier {
            switch identifier {
            case "TangLoginNaviCtrlSegueID":
                loginNaviCtrl = segue.destination as! TangLoginNaviCtrl
                loginNaviCtrl.didLogined = {
                    [unowned self] in
                    self.logined()
                }
            default: break
            }
        }
    }
}
